packages/minimongo/common.js

```diff
@@ -87,8 +87,41 @@ export const ELEMENT_OPERATORS = {
     // should *not* include it itself.
     dontIncludeLeafArrays: true,
     compileElementSelector(operand) {
-      if (typeof operand !== 'number') {
-        throw Error('$type needs a number');
+      if (typeof operand === 'string') {
+        const operandAliasMap = {
+          'double': 1,
+          'string': 2,
+          'object': 3,
+          'array': 4,
+          'binData': 5,
+          'undefined': 6,
+          'objectId': 7,
+          'bool': 8,
+          'date': 9,
+          'null': 10,
+          'regex': 11,
+          'dbPointer': 12,
+          'javascript': 13,
+          'symbol': 14,
+          'javascriptWithScope': 15,
+          'int': 16,
+          'timestamp': 17,
+          'long': 18,
+          'decimal': 19,
+          'minKey': -1,
+          'maxKey': 127,
+        };
+        if (!operandAliasMap.hasOwnProperty(operand)) {
```

Comment:
In order to protect against the (unlikely, but seemingly more likely than usual case, given that this is a list of operands) scenario where operands includes a property called `hasOwnProperty`, could we switch this to?

```js
Object.prototype.hasOwnProperty.call(operandAliasMap, operand);
```

Reply:
yep, apparently there was already a `hasOwn` shorthand foreseen.

Reply:
Yeah, we frequently opt to use that technique! Always feel free to use it as you see fit! :smile: